presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java

```diff
@@ -139,6 +139,36 @@ public int getSetBits(int batchSize, boolean[] vector)
         return count;
     }
 
+    // vector[i] is set to the bit at offsets[i] - offsetBase for i
+    // from 0 to numOffsets. The offset[i] - offsetBase values are
+    // relative to the current position of the stream. 0 would mean
+    // the value returned by a call to next(). The stream is advanced
+    // by numBits. The offset values are increasing and non-repeating.
+    public void getSetBits(int[] offsets, int numOffsets, int offsetBase, int numBits, boolean[] vector)
+            throws IOException
+    {
+        if (offsets[0] == offsetBase && offsets[numOffsets - 1] == offsetBase + numOffsets - 1) {
+            getSetBits(numOffsets, vector);
+            if (numBits > offsets[numOffsets - offsetBase - 1] + 1) {
+                skip(numBits - offsets[numOffsets - 1] - offsetBase - 1);
+            }
+            return;
+        }
+        int position = 0;
```

Comment:
target and position names are not very descriptive; let's rename to currentOffset and offset:

```
        int currentOffset = 0;
        for (int i = 0; i < numOffsets; i++) {
            int offset = offsets[i] - offsetBase;
            if (offset > currentOffset) {
                skip(offset - currentOffset);
                currentOffset = offset;
            }
            vector[i] = nextBit();
            currentOffset++;
        }
        if (numBits > offsets[numOffsets - 1] - offsetBase) {
            skip(numBits - (offsets[numOffsets - 1] - offsetBase));
        }
```